applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -94,38 +87,26 @@ static struct bt_audio_discover_params
 static int discover_source(struct bt_conn *conn);
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 
-static int next_free_audio_stream_index_get(uint8_t *index)
-{
-	for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-		if (audio_streams[i].conn == NULL) {
-			*index = i;
-			return 0;
-		}
-	}
-	LOG_WRN("All stream spots taken");
-	return -ENOSPC;
-}
-
 /**
- * @brief  Get audio stream index based on the stream pointer
+ * @brief  Check if an endpoint is in the given state
  *
- * @param  stream  The stream to search for
- * @param  index   The audio_stream index
+ * @param[in]  ep   The endpoint to check
+ * @param state     The state to check for
  *
- * @return 0 if succesfull, -EINVAL if there is no match
+ * @return True if the endpoint is in the given state, false otherwise
  */
-static int audio_stream_index_get(struct bt_audio_stream *stream, uint8_t *index)
+static bool ep_state_check(struct bt_audio_ep *ep, enum bt_audio_state state)
 {
-	for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-		if (&audio_streams[i] == stream) {
-			*index = i;
-			return 0;
-		}
+	if (ep == NULL) {
+		LOG_WRN("ep is NULL");
```

Comment:
Should we put this into LOG_DBG?

Reply:
Changed back to LOG_DBG, and made a @note for the ep_state_check() function